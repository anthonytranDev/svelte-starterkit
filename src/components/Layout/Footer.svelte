<style>
  .footer {
    padding: 1rem 0.75rem;
  }

  .footer-wrapper {
    display: flex;
    justify-content: space-between;
  }

  .footer-license {
    font-weight: 700;
  }

  #github-logo {
    position: relative;
    top: 3px;
    height: 20px;
  }
</style>

<div class="footer" data-testid="Footer">
  <div class="footer-wrapper">
    <div class="footer-license">MIT License</div>
    <div>
      <a href="https://github.com/anthonytranDev">
        <img
          id="github-logo"
          src="/github-logo.svg"
          class="home-logo"
          alt="logo"
        />
        <span>GitHub: anthonytranDev/svelte-basic-starter</span>
      </a>
    </div>
  </div>
</div>
