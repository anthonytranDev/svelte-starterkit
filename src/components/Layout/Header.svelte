<script>
  import { Link } from 'svelte-routing'
</script>

<style>
  span {
    font-weight: 700;
  }

  .header {
    padding: 1rem 0.5rem;
  }

  .header-wrapper {
    display: flex;
    justify-content: space-between;
  }

  .header__nav {
    display: inline-flex;
  }

  .header-item {
    padding: 1rem 0.75rem;
  }
</style>

<div class="header" data-testid="Header">
  <div class="header-wrapper">
    <Link to="/">
      <div class="header-item">
        <span>Basic</span> Starterkit
      </div>
    </Link>

    <nav class="header__nav">
      <Link to="/">
        <div class="header-item">Home</div>
      </Link>
      <Link to="/about">
        <div class="header-item">About</div>
      </Link>
      <Link to="/blog">
        <div class="header-item">Blog</div>
      </Link>
    </nav>
  </div>
</div>
