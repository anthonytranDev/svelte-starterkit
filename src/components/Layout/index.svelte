<script>
  import Footer from './Footer.svelte'
  import Header from './Header.svelte'
</script>

<style>
  .layout-background {
    background-color: #f9f6f6;
    color: #333;
    min-height: 100vh;
    padding: 1rem;
  }
</style>

<div data-testid="Layout">
  <Header />
  <div class="layout-background">
    <slot />
  </div>
  <Footer />
</div>
