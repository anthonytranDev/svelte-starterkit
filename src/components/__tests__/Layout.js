import "@testing-library/jest-dom/extend-expect"
import { render } from "@testing-library/svelte"
import { Router } from "svelte-routing"

import Layout from "../Layout/index.svelte"

describe("Layout", () => {
  it("should render", () => {
    const { getByTestId } = render(
      <Router>
        <Layout />
      </Router>
    )

    expect(getByTestId("Layout")).toMatchSnapshot()
  })
})
