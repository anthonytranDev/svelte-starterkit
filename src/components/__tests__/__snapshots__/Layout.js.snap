// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout should render 1`] = `
<div
  data-testid="Layout"
>
  <div
    class="header svelte-1l7lnf1"
    data-testid="Header"
  >
    <div
      class="header-wrapper svelte-1l7lnf1"
    >
      <a
        aria-current="page"
        href="/"
      >
        <div
          class="header-item svelte-1l7lnf1"
        >
          <span
            class="svelte-1l7lnf1"
          >
            Basic
          </span>
           Starterkit
        </div>
      </a>
       
      <nav
        class="header__nav svelte-1l7lnf1"
      >
        <a
          aria-current="page"
          href="/"
        >
          <div
            class="header-item svelte-1l7lnf1"
          >
            Home
          </div>
        </a>
         
        <a
          href="/about"
        >
          <div
            class="header-item svelte-1l7lnf1"
          >
            About
          </div>
        </a>
         
        <a
          href="/blog"
        >
          <div
            class="header-item svelte-1l7lnf1"
          >
            Blog
          </div>
        </a>
      </nav>
    </div>
  </div>
   
  <div
    class="layout-background svelte-1a5pxw2"
  />
   
  <div
    class="footer svelte-ds2ccu"
    data-testid="Footer"
  >
    <div
      class="footer-wrapper svelte-ds2ccu"
    >
      <div
        class="footer-license svelte-ds2ccu"
      >
        MIT License
      </div>
       
      <div>
        <a
          href="https://github.com/anthonytranDev"
        >
          <img
            alt="logo"
            class="home-logo svelte-ds2ccu"
            id="github-logo"
            src="/github-logo.svg"
          />
           
          <span>
            GitHub: anthonytranDev/svelte-basic-starter
          </span>
        </a>
      </div>
    </div>
  </div>
</div>
`;
